### PR TITLE
Make SupportedVersions respect SSL_OP_NO_TLSv*

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: src/tls.c [hunk 1]
```diff
@@ -5103,44 +5118,65 @@ static int TLSX_SupportedVersions_Write(void* data, byte* output,
...
+            if ((ssl->options.mask & SSL_OP_NO_TLSv1_1) == 0)
+        #endif
+            {
+                *cnt += OPAQUE16_LEN;
+                *(output++) = major;
+                *(output++) = (byte)TLSv1_1_MINOR;
+            }
+    #ifdef WOLFSSL_ALLOW_TLSV10
+        #if defined(OPENSSL_EXTRA) || defined(HAVE_WEBSERVER)
+            if ((ssl->options.mask & SSL_OP_NO_TLSv1_0) == 0)
```
Comment (kaleb-himes) [inline src/tls.c]: ```SSL_OP_NO_TLSv1_0``` should just be ```SSL_OP_NO_TLSv1```

File: src/tls.c [hunk 2]
```diff
@@ -5066,19 +5066,34 @@ static int TLSX_SupportedVersions_GetSize(void* data, byte msgType, word16* pSz)
...
-        cnt++;
+        #if defined(OPENSSL_EXTRA) || defined(HAVE_WEBSERVER)
+            if ((ssl->options.mask & SSL_OP_NO_TLSv1_1) == 0)
+        #endif
+                cnt++;
     #ifdef WOLFSSL_ALLOW_TLSV10
-        /* TLS v1.0  */
-        cnt++;
+        #if defined(OPENSSL_EXTRA) || defined(HAVE_WEBSERVER)
+            if ((ssl->options.mask & SSL_OP_NO_TLSv1_0) == 0)
```
Comment (kaleb-himes) [inline src/tls.c]: ```SSL_OP_NO_TLSv1_0``` should just be ```SSL_OP_NO_TLSv1```

**Comment by kaleb-himes** [comment]
Thanks for the quick patch sean, looks great with only one thing I noticed based on my tests from yesterday:

```
// KH
unsigned long mask;
long myRet = 0;
mask = SSL_OP_NO_TLSv1;
myRet = wolfSSL_set_options(ssl, mask);
printf("myRet = %ld", myRet);
```

Notice the mask I was testing was ```SSL_OP_NO_TLSv1``` but it looks like you were using the macro ```SSL_OP_NO_TLSv1_0```. Either we should change the enumeration in ssl.h from ```SSL_OP_NO_TLSv1``` to include the ```_0``` postfix or we should use the existing enum. (I think the existing enum is for the compatibility layer so we should probably stick with the existing enum.

Cheers,

K

File: src/tls.c [hunk 3]
```diff
@@ -5103,44 +5118,65 @@ static int TLSX_SupportedVersions_Write(void* data, byte* output,
...
+                *(output++) = (byte)TLSv1_1_MINOR;
+            }
+    #ifdef WOLFSSL_ALLOW_TLSV10
+        #if defined(OPENSSL_EXTRA) || defined(HAVE_WEBSERVER)
+            if ((ssl->options.mask & SSL_OP_NO_TLSv1_0) == 0)
+        #endif
+            {
+                *cnt += OPAQUE16_LEN;
+                *(output++) = major;
+                *(output++) = (byte)TLSv1_0_MINOR;
```
Comment (kaleb-himes) [inline src/tls.c]: Oh, when trying to build the branch I get an error on this one too, I think this should be enum: ```TLSv1_MINOR```

**Comment by kaleb-himes** [comment]
With the above noted changes I can confirm the tests I have setup now look correct and when disabling the versions I no longer see them in the extension.

**Comment by kaleb-himes** [comment]
Given that the above errors were not caught in the PR tests I'll add a test config which I'll put up for your review sean to include:

```
./configure --enable-opensslextra --enable-oldtls --enable-sslv3 --enable-tlsv10 --enable-tls13
```

Which is what was used for testing this case.

File: src/tls.c (hunk 1 above)
Comment (SparkiDev) [reply src/tls.c]: fixed

File: src/tls.c (hunk 2 above)
Comment (SparkiDev) [reply src/tls.c]: fixed

File: src/tls.c (hunk 3 above)
Comment (SparkiDev) [reply src/tls.c]: fixed

**Pull request merged**